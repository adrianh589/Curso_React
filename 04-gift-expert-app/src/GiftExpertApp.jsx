import {useState} from "react";
import {AddCategory, GifGrid} from "./components/index.js";

export const GiftExpertApp = () => {
    const [categories, setCategories] = useState(['One Punch']);

    const onAddCategory = (newCategory) => {
        console.log('Hola, soy el onAddCategory******')
        console.log(newCategory);
        // setCategories([newCategory, ...categories]); // Mi forma
        if (categories.includes(newCategory)) return;
        setCategories(cat => [newCategory, ...cat]); // Segunda Forma
    }

    return (
        <>
            {/*Titulo*/}
            <h1>GiftExpertApp</h1>

            {/*  Input  */}
            <AddCategory
                // setCategories={setCategories}
                onNewCategory={value => onAddCategory(value)}
            />

            {
                categories.map((category) => (
                    <GifGrid key={category} category={category}/>
                ))
            }
        </>
    )
}