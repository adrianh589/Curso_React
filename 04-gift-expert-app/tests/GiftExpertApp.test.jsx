import {fireEvent, render, screen} from "@testing-library/react";
import {GiftExpertApp} from "../src/GiftExpertApp.jsx";
import {GiftItem} from "../src/components/index.js";

describe('Pruebas en <GiftExpertApp />', () => {

    /**
     * TODO:
     * 1. Posteo del formulario y ver que pasa (snapshot)
     * 2. Que pasa si mando la misma categoria
     * 3. Que pasa si mando una categoria diferente
     */

    test('Debe de tomar el snapshot', () => {
        const {container} = render( <GiftExpertApp /> );
        screen.debug();
        expect(container).toMatchSnapshot();
    });

    test('Debe de mandar una categoria', () =>{
        render( <GiftExpertApp /> );

        const form = screen.getByRole('form');
        const input = screen.getByRole('textbox');

        console.log(form);

        fireEvent.input(input, {target: {value: 'De prueba'}});
        fireEvent.submit(form);

        screen.debug();
    });
});