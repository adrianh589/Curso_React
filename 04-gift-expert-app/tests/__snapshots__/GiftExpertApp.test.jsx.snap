// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftExpertApp /> Debe de tomar el snapshot 1`] = `
<div>
  <h1>
    GiftExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Agregar
    </button>
  </form>
  <h3>
    One Punch
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
